style_guides/html_style_guide.md

```diff
@@ -1,6 +1,31 @@
-
 # HTML Style Guide
 
+## Camel case ID and other attribute values
+
+Use camel case for the values of attributes such as IDs and data test subject selectors.
+
+```html
+<button
+  id="veryImportantButton"
+  data-test-subj="clickMeButton"
+>
+  Click me
+</button>
+```
+
+The only exception is in cases where you're dynamically creating the value, and you need to use
+hyphens as delimiters:
+
+```html
+<button
+  ng-repeat="button in buttons"
+  id="{{ 'veryImportantButton-' + button.id }}"
+  data-test-subj="{{ 'clickMeButton-' + button.id }}"
+>
+  {{ button.label }}
+</button>
```

Comment:
Per Slack convo w/ @archanid, we could expand this exception to use hyphens as a grouping mechanism among similar elements, e.g.

```html
<section id="welcomePage-introSection">
  <!-- Content -->
</section>

<section id="welcomePage-aboutSection">
  <!-- Content -->
</section>
```

Ater thinking about this a bit I'm not sure how useful this would be, and I'd prefer to keep our naming as simple as possible. This still seems readable and understandable to me:

```html
<section id="welcomePageIntroSection">
  <!-- Content -->
</section>

<section id="welcomePageAboutSection">
  <!-- Content -->
</section>
```